doc/memos/rdm-draft-alamos-lanzieri-runtime-configuration-architecture.md

```diff
@@ -0,0 +1,139 @@
+- RDM: rdm-draft-alamos-lanzieri-runtime-configuration-architecture.md
+- Title: Runtime configuration architecture
+- Authors: José Álamos, Leandro Lanzieri
+- Status: draft
+- Type: Design
+- Created: December 2018
+
+## Abstract
+This memo describes the proposed high-level architecture and
+mechanisms to implement a runtime configuration system on a RIOT node.
+
+A runtime configuration system is in charge of providing a mechanism to set and
+get the values of configuration parameters that are used during the execution of the firmware,
+as well as a way to persist these values. Most of times these values are
+per-node specific.
+
+Examples of runtime configurations are:
+- Transmission duty cycles
+- Sensor thresholds
+- Private-public keys
+- System state variables
+
+These parameters might have constraints, like an specific order to be applied
+(due to interdependencies) or value boundaries.
+
+The main advantages of having such a system are:
+- Easy to apply per-node configuration during deployment
+- No need to implement a special mechanism for per-node configurations during
+  firmware updates (only in the case of migration)
+- Common interface for modules to expose their runtime configuration parameters and handle
+  them
+- Transparent interface for storing configuration parameters in non-volatile
+  storage devices
+
+## Status
+This document is currently under open discussion. This document is a product of
+[Configuration Task
+Force](https://github.com/RIOT-OS/RIOT/wiki/Configuration-Task-Force-(CTF)), and
+aims to describe the architecture of a runtime configuration system. The content
+of this document is licensed with a Creative Commons CC-BY-SA license.
+
+## Terminology
+This memo uses the [RFC2119](https://www.ietf.org/rfc/rfc2119.txt) terminology
+and the following acronyms:
+
+- RDM: RIOT Developer Memo
+- CTF: RIOT's Configuration Task Force
+- RCS: Runtime Configuration System
+
+# 1. Introduction
+This document specifies the proposed architecture by the
+Configuration Task Force (CFT) to implement a secure and reliable Runtime
+Configuration System (RCS), focusing in modularity, reusing of existing
+technologies in RIOT (network stack, storage interface) and adhering to
+standards.
+
+# 2. Architecture
+The RCS is formed by [the RIOT Registry](3-the-riot-registry), one or more [Registry Handlers](4-registry-handlers), one or more
+[storage facilities](5-storage-facilities) and a [configuration manager](6-configuration-manager) with one or more interfaces. The
+runtime configuration manager could in the future be replaced by a generic
+manager.  ![](./files/rdm-draft-alamos-lanzieri-runtime-configuration-architecture/architecture.png "Runtime Configuration Architecture")
+
+# 3. The RIOT Registry
+The RIOT Registry exposes an API for handling 
+**persistent module configurations**. The RIOT registry
+allows, among others, to:
```

Comment:
"among others" is not very helpful: could you give a complete list of operations?